### PR TITLE
Replace request internals with fetch

### PR DESCRIPTION
**Pull request by southpolesteve**

### CODE REVIEW

File: src/retry/retryUtility.ts [hunk 1]
```diff
@@ -110,45 +91,43 @@ export async function apply(
...
     if (!results) {
-      headers[Constants.ThrottleRetryCount] = resourceThrottleRetryPolicy.currentRetryAttemptCount;
-      headers[Constants.ThrottleRetryWaitTimeInMs] = resourceThrottleRetryPolicy.cummulativeWaitTimeinMilliseconds;
+      headers[Constants.ThrottleRetryCount] = retryPolicies.resourceThrottleRetryPolicy.currentRetryAttemptCount;
+      headers[Constants.ThrottleRetryWaitTimeInMs] =
+        retryPolicies.resourceThrottleRetryPolicy.cummulativeWaitTimeinMilliseconds;
       err.headers = { ...err.headers, ...headers };
       throw err;
     } else {
       request.retryCount++;
```
Comment (christopheranderson) [inline src/retry/retryUtility.ts]: It looks like newUrl not being used is a bug. I think I broke in the commit after this one: https://github.com/Azure/azure-cosmos-js/blame/b22194817b958aa117205c059b513c9a54f413f3/src/retry/retryUtility.ts#L120-L122

File: src/request/RequestHandler.ts [hunk 2]
```diff
@@ -19,9 +24,73 @@ export class RequestHandler {
...
+        method: requestOptions.method,
+        headers: requestOptions.headers as any,
+        agent: requestOptions.agent,
+        signal,
+        ...(body && { body })
+      } as any); // TODO Remove any. Upstream issue https://github.com/lquixada/cross-fetch/issues/42
+    } catch (error) {
+      if (error.name === "AbortError") {
+        if (didTimeout === true) {
+          throw new TimeoutError();
```
Comment (christopheranderson) [inline src/request/RequestHandler.ts]: We currently expect this to throw ECONNRESET when we successfully hungup, then we try again, so we need to figure out how to handle retries. I need to look into how cross-fetch works in this case.

File: src/LocationCache.ts [hunk 3]
```diff
@@ -115,7 +115,8 @@ export class LocationCache {
     // then default to the first two write locations, alternating (or the default endpoint)
     if (
       request.locationRouting.ignorePreferredLocation ||
-      (!isReadRequest(request) && !this.canUseMultipleWriteLocations(request))
+      (!isReadRequest(request.operationType) &&
+        !this.canUseMultipleWriteLocations(request.resourceType, request.operationType))
```
[the remark this answers is not in the document]
Comment (southpolesteve) [reply src/LocationCache.ts]: Yeah just prettier trying to make things pretty

File: src/request/RequestHandler.ts [hunk 4]
```diff
@@ -156,27 +190,10 @@ export class RequestHandler {
       this.globalEndpointManager,
       this.connectionPolicy,
       this.requestAgent,
-      "PUT",
+      HTTPMethod.put,
       urlString,
       request,
       body,
```
[the remark this answers is not in the document]
Comment (southpolesteve) [reply src/request/RequestHandler.ts]: It's not used. Not 100% sure why I removed it though. Can add back if you want.

**Pull request merged**